core/lib/protobuf_config/src/consensus.rs

```diff
@@ -100,14 +100,28 @@ impl ProtoRepr for proto::Config {
             let addr = Host(required(&e.addr).context("addr")?.clone());
             anyhow::Ok((key, addr))
         };
+
+        let max_payload_size = required(&self.max_payload_size)
+            .and_then(|x| Ok((*x).try_into()?))
+            .context("max_payload_size")?;
+
+        let max_batch_size = match self.max_batch_size {
+            Some(x) => x.try_into().context("max_batch_size")?,
+            None => {
+                // Compute a default batch size: the batch interval is ~1 minute,
+                // so there will be ~60 blocks, and an Ethereum Merkle proof is ~1kB.
+                // Using 100 to be generous.
+                max_payload_size * 100 + kB
+            }
+        };
```

Comment:
Thanks for picking up on this, I wasn't sure what to do. I don't know how disruptive it is for operators to have to add a new field in their configs and pick a good value, at a time when the feature isn't even fully developed yet. I thought it might be easier for us to start like this, discuss what it should be, and if we get it wrong it can always be set explicitly. 

Ultimately I just didn't want anyone not to be able to start their nodes because of this. We can do whatever you guys think is the best way to add it. 

You are saying the number of blocks in a batch is highly variable, but is that because during high transaction volume there are more batches, and during quiet period they are less frequent? The `max_payload_size` is a hard limit on the block size, so I'd assume if there are loads of transactions the blocks will be close to this value in size, and batches will have fewer than (say) 100 of them. Conversely if there aren't many transactions, there might be more than 100 blocks, but they will be nowhere near the `max_payload_size` individually. 

I suppose by making it required we have to recommend something along similar lines. Or we could take an actual setting for the maximum blocks in a batch, if it exists at all.

Reply:
> I thought it might be easier for us to start like this, discuss what it should be, and if we get it wrong it can always be set explicitly.

> Ultimately I just didn't want anyone not to be able to start their nodes because of this. We can do whatever you guys think is the best way to add it.

Sounds good to me!

re: batch size.
batches may contain thousands of transactions, and in extreme case every transaction may take one l2 block (since they are closed every second - imagine constant 1TPS load - one transcation == one l2 block), so `* 100` may be not enough. Such 1-transaction l2 blocks are likely to be tiny - but not guaranteed - we may have transactions with a lot of calldata. Can we use much higher value here? ` * 5000 ` perhaps?

Reply:
Sure, so just to be clear that's 5000x the maximum _block_ size, not not transaction size.

Reply:
Well, they fill entire block not necessarily because they are large but simply because we close blocks every second. However to achieve this extreme case that would stretch l1 batches, they also need to be large, indeed.

does this clarify? We can discuss further or potentially huddle if you like.

overall we need a high enough value that would definitely fit any l1 batch size, so we need to be very aggressive here sand only set very high limits. But isn’t there a trade off? E.g. dos suspectabilliy?

Reply:
Yes, the tradeoff is DoS prevention. It's used in two places:
* Limits the response size when our node [pulls](https://github.com/matter-labs/era-consensus/blob/1c20ca037a810edbb02f063567f9a99d41bb803c/node/actors/network/src/gossip/runner.rs#L359) the `SyncBatch` from peers
* Limits the request size when peers [push](https://github.com/matter-labs/era-consensus/blob/1c20ca037a810edbb02f063567f9a99d41bb803c/node/actors/network/src/gossip/runner.rs#L118) their `BatchStoreState` to us, which contains a `SyncBatch` (although I'm not sure this is necessary, since all we seem to use it for is to gate requests we send - maybe we shortcut pulls if it already has it, it's [complicated](https://github.com/matter-labs/era-consensus/blob/1c20ca037a810edbb02f063567f9a99d41bb803c/node/actors/network/src/gossip/fetch.rs#L149))

Reply:
Changed it to 5000 👍

Reply:
> I don't know how disruptive it is for operators to have to add a new field in their configs and pick a good value, at a time when the feature isn't even fully developed yet.

I wouldn't have them pick the value. I would just have the value set as default in the config. AFAIK, that's what we do with the `max_payload_size`.

Reply:
Ah, okay, that would be great! I didn't dig deep enough to understand if there is a default config that gets rolled out with new releases 👀

Reply:
I found these two places where consensus config is read from YAML files:
* for [external nodes](https://github.com/matter-labs/zksync-era/blob/f4410e3254dafdfe400e1c2c420f664ba951e2cd/core/bin/external_node/src/config/mod.rs#L1141-L1146)
* for [the server](https://github.com/matter-labs/zksync-era/blob/f4410e3254dafdfe400e1c2c420f664ba951e2cd/core/bin/zksync_server/src/config.rs#L18)

In both cases the method returns an `Option` but the caller would [error out](https://github.com/matter-labs/zksync-era/blob/f4410e3254dafdfe400e1c2c420f664ba951e2cd/core/bin/zksync_server/src/main.rs#L157) if the config is missing. 

The only potential default setting I see is [consensus_config.yaml](https://github.com/matter-labs/zksync-era/blob/main/etc/env/consensus_config.yaml) which has some localhost-only entries that they say shouldn't be copied. I can add a value there, but this file doesn't look like the entries are going to be copied anywhere else, such as Kubernetes configs.

Where did you think the defaults should be added?